app/scripts/metamask-controller.js

```diff
@@ -7797,10 +7792,11 @@ export default class MetamaskController extends EventEmitter {
       networkClientId,
       txParams: { from },
     } = transactionMeta;
-    this.accountTrackerController.updateAccountByAddress({
-      address: from,
-      networkClientId,
-    });
+    // TODO AccountTrackerController Migration: Review how we handle this
+    // this.accountTrackerController.updateAccountByAddress({
+    //   address: from,
+    //   networkClientId,
+    // });
```

Comment:
There is no equivalent method in this version of the controller to update a specific account for a specific network.

- `refresh` does not work with a specific account, but with the selected account, which is not good here as the transaction could be for any account in the wallet.
- `syncBalanceWithAddresses` only returns the results, but it does not update the state.

We can either do:
- Full refresh of all accounts for that network (unnecessarily expensive)
- No refresh at all and wait for the balances to update on their own
- Make changes to `AccountTrackerController` to support this feature